pkg/counter/counter_test.go

```diff
@@ -9,45 +9,87 @@ import (
 )
 
 //
-// Increase
+// Increment
 //
 
-// TestZeroedIncrease tests that the counter is not increased when 0 is given.
-func TestZeroedIncrease(t *testing.T) {
+func TestIncrement(t *testing.T) {
 	expected := uint64(1)
-	c := NewCounter(1)
+	c := NewCounter(0)
 
-	err := c.Increase(0)
+	err := c.Increment()
 
 	require.NoError(t, err)
-	require.Equal(t, expected, c.Read())
+	require.Equal(t, expected, c.Get())
 }
 
-// TestIncreaseAndRead tests that the counter is increased by 1 and returns the new value.
-func TestIncreaseAndRead(t *testing.T) {
-	expected := uint64(1)
+func TestIncrementWithValue(t *testing.T) {
+	expected := uint64(9)
 	c := NewCounter(0)
 
-	n, err := c.IncreaseAndRead()
+	err := c.Increment(9)
 
 	require.NoError(t, err)
-	require.Equal(t, expected, n)
-	require.Equal(t, expected, c.Read())
+	require.Equal(t, expected, c.Get())
 }
 
-// TestIncrease tests that the counter is increased by 1.
-func TestIncrease(t *testing.T) {
-	expected := uint64(1)
+func TestIncrementWithMultipleValue(t *testing.T) {
+	expected := uint64(9)
 	c := NewCounter(0)
 
-	err := c.Increase()
+	err := c.Increment(3, 3, 3)
 
 	require.NoError(t, err)
-	require.Equal(t, expected, c.Read())
+	require.Equal(t, expected, c.Get())
 }
 
-// TestIncrease_MultipleThreads is a stress test to check thread safety for increased.
-func TestIncrease_MultipleThreads(t *testing.T) {
+func TestIncrementAndRead(t *testing.T) {
+	expected := uint64(9)
+	c := NewCounter(0)
+
+	n, err := c.IncrementValueAndRead(9)
+
+	require.NoError(t, err)
+	require.Equal(t, expected, n)
+	require.Equal(t, expected, c.Get())
+}
+
+func TestZeroedIncrementWithValue(t *testing.T) {
+	expected := uint64(9)
+	c := NewCounter(9)
+
+	err := c.Increment(0)
```

Comment:
```suggestion
	err := c.Increment(0, 0, 0)
```

Reply:
Im adding this test AND benchmark tests.